go/libraries/doltcore/sqle/statspro/dolt_stats.go

```diff
@@ -26,54 +26,158 @@ import (
 )
 
 type DoltStats struct {
-	mu *sync.Mutex
+	Statistic *stats.Statistic
+	mu        *sync.Mutex
 	// Chunks is a list of addresses for the histogram fanout level
 	Chunks []hash.Hash
 	// Active maps a chunk/bucket address to its position in
 	// the histogram. 1-indexed to differentiate from an empty
 	// field on disk
 	Active map[hash.Hash]int
+	Hist   sql.Histogram
+}
+
+var _ sql.Statistic = (*DoltStats)(nil)
+
+func (s *DoltStats) WithColSet(set sql.ColSet) sql.Statistic {
```

Comment:
I'm not crazy about all this boilerplate code to avoid an embedding but I guess it's slightly preferable to this breaking silently when you add a new `With` method to the interface type

Reply:
the embedded version i started with did have confusing looking errors for missed Withs, so i switched to this one